airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryDestination.kt

```diff
@@ -83,23 +95,194 @@ class BigQueryDestination : BaseConnector(), Destination {
+                            hasUnprocessedRecords = true,
+                            maxProcessedTimestamp = Optional.empty()
+                        ),
+                    isSchemaMismatch = true,
+                    isFinalTableEmpty = true,
+                    destinationState = BigQueryDestinationState(needsSoftReset = false)
+                )
+
+            // We simulate a mini-sync to see the raw table code path is exercised. and disable T+D
+            destinationHandler.createNamespaces(setOf(defaultDataset, rawDatasetOverride))
+
+            val bigQueryGcsStorageOperations =
+                BigQueryGcsStorageOperation(
+                    gcsOperations,
+                    gcsConfig,
+                    gcsNameTransformer,
+                    keepStagingFiles,
+                    bigquery,
+                    sqlGenerator,
+                    destinationHandler,
+                )
+
+            val streamOperation: StagingStreamOperations<BigQueryDestinationState> =
+                StagingStreamOperations(
+                    bigQueryGcsStorageOperations,
+                    initialStatus,
+                    FileUploadFormat.CSV,
+                    V2_WITH_GENERATION,
+                    disableTypeDedupe = true
+                )
+
+            // Dummy message
+            val data =
+                """
+                {"testKey": "testValue"}
+                 """.trimIndent()
+
+            val message =
+                PartialAirbyteMessage()
+                    .withSerialized(data)
+                    .withRecord(
+                        PartialAirbyteRecordMessage()
+                            .withEmittedAt(System.currentTimeMillis())
+                            .withMeta(
+                                AirbyteRecordMessageMeta(),
+                            ),
+                    )
+
+            streamOperation.writeRecords(streamConfig, listOf(message).stream())
+            streamOperation.finalizeTable(
+                streamConfig,
+                StreamSyncSummary(1, AirbyteStreamStatusTraceMessage.AirbyteStreamStatus.COMPLETE),
+            )
+
+            // clean up the raw table, this is intentionally not part of actual sync code
+            // because we avoid dropping original tables directly.
+            destinationHandler.execute(
+                Sql.of(
+                    "DROP TABLE IF EXISTS $projectId.${streamId.rawNamespace}.${streamId.rawName};",
+                ),
+            )
+
+            return AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED)
         } catch (e: Exception) {
-            log.error(e) { "Check failed." }
+
+            log.error(e) { "checkGcsCopyPermission failed." }
+
             throw ConfigErrorException((if (e.message != null) e.message else e.toString())!!)
+        } finally {
+
+            try {
+                // In the finally block, check again to clean up the raw table
```

Comment:
why do we need to do this on line 259 then? Isn't doing it in the finally block enough?

Reply:
sure, this is a good observation, I have removed the clean up code in the try block and kept it only in the finally block.